lib/features/steps/request_assertion_steps.rb

```diff
@@ -55,6 +55,16 @@
   assert_equal(header_value, Server.current_request[:request][header_name])
 end
 
+# Tests that a header matches a regex
+#
+# @step_input header_name [String] The header to test
+# @step_input regex_string [String] The regex to match with
+Then('the {string} header matches the regex {string}') do |header_name, regex_string|
+  regex = Regexp.new(regex_string)
+  value = Server.current_request[:request][header_name]
```

Comment:
Is there a case for asserting that the header exists first, just for a friendlier message if it fails?